File: app/src/main/java/com/mir/panosdev/cookingrecipesmvp/mvp/model/ingredient/IngredientsResponse.java
package com.mir.panosdev.cookingrecipesmvp.mvp.model.ingredient;

public class IngredientsResponse implements java.io.Serializable {
    private Ingredient[] ingredients;

    public Ingredient[] getIngredients() {
        return this.ingredients;
    }

    public void setIngredients(Ingredient[] ingredients) {
        this.ingredients = ingredients;
    }
}

File: app/src/main/java/com/mir/panosdev/cookingrecipesmvp/mvp/model/ingredient/IngredientsResponse.kt
package com.mir.panosdev.cookingrecipesmvp.mvp.model.ingredient

import java.util.*

data class IngredientsResponse(var ingredients: Array<Ingredient>) : java.io.Serializable {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as IngredientsResponse

        if (!Arrays.equals(ingredients, other.ingredients)) return false

        return true
    }

    override fun hashCode(): Int {
        return Arrays.hashCode(ingredients)
    }
}
